### PR TITLE
react 0.14, react-router 1.0.0-rc3

### PR DESCRIPTION
**Pull request by ngokevin**
use action creators to do pushStates rather than using components to trigger transitions using context.router

### CODE REVIEW

**Comment by ngokevin** [comment]
need to fix reviewer search. my reverse lib might not be handling params well

File: src/media/js/addon/components/search.js [hunk 1]
```diff
@@ -37,12 +37,14 @@ export default class Search extends React.Component {
 
         {this.props.results.length && this.state.q &&
           <ul className="search-results">
-            {this.props.results.map(result =>
-              <li className="search-result">
-                <ReverseLink to="addon-review-detail"
-                             params={{slug: result.slug}}>
-                  {result.name} ({result.slug})
-                </ReverseLink>
+            {this.props.results.map((result, i) =>
```
Comment (chuckharmston) [inline src/media/js/addon/components/search.js]: This is really painful to read. Can you move the function into a render method so you can oneline it?

File: src/media/js/routes.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
```
Comment (chuckharmston) [inline src/media/js/routes.js]: Glad we did this.

**Comment by chuckharmston** [comment]
Love the radio of :heavy_plus_sign: to :heavy_minus_sign: in here.

**Comment by chuckharmston** [comment]
Generally, everything looks good. Make sure you test the TOS workflow thoroughly, feels like it should require more code change than this.

**Comment by chuckharmston** [comment]
r+

File: src/media/js/routes.js (hunk 2 above)
Comment (ngokevin) [inline src/media/js/routes.js]: couldn't really do it before since it would've caused a circular dep

**Comment by ngokevin** [comment]
tos wfm with some code stubbing

**Comment by ngokevin** [comment]
thanks

**Pull request merged**